### PR TITLE
MAT-6785 SimpleXML & HQMF generation, round 2

### PR DESCRIPTION
**Pull request by adongare**
- CQL lookup mappings
- Fetch CQL lookups
- HQMF generation
- Exception handling

## MADiE PR

Jira Ticket: [MAT-6785](https://jira.cms.gov/browse/MAT-6785)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/services/PackagingService.java [hunk 1]
```diff
@@ -44,9 +44,18 @@ public byte[] createMeasurePackage(Measure measure, String accessToken) {
...
-    entries.put(
-        measure.getEcqmTitle() + "-v" + measure.getVersion() + "-QDM" + ".xml", hqmf.getBytes());
+    // TODO: remove this after HQMF is complete, but for now don't prevent export if HQMF fails
+    try {
+      String hqmf = hqmfService.generateHqmf((QdmMeasure) measure, accessToken);
+      entries.put(
+          measure.getEcqmTitle() + "-v" + measure.getVersion() + "-QDM" + ".xml", hqmf.getBytes());
+    } catch (Exception ex) {
+      // TODO: this is temporary - remove it after!!
+      log.error("An error occurred during HQMF generation!", ex);
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/services/PackagingService.java]: Adding a measure ID in the log statements will help us finding issues on Splunk

File: src/main/java/gov/cms/madie/resources/PackageController.java [hunk 2]
```diff
@@ -51,10 +54,13 @@ public byte[] getMeasurePackage(
       },
       consumes = {MediaType.APPLICATION_JSON_VALUE})
   public String getMeasureSimpleXml(
-      @RequestBody @Validated(Measure.ValidationSequence.class) Measure measure)
+      @RequestBody @Validated(Measure.ValidationSequence.class) Measure measure,
+      @RequestHeader("Authorization") String accessToken)
       throws JAXBException {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/resources/PackageController.java]: I am unable to get a full picture of who is the client of "/simple-xml" resource, but do you guys think throwing an exception from Controller is ok ? I just feel like we shouldn't be in a situation where something is failing and we have no idea, because we are not even logging it

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/resources/PackageController.java]: We can ignore it if not important for this release

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/madie/resources/PackageController.java (hunk 2 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/resources/PackageController.java]: That's a good point - we should have that caught by the controller advice. I don't think any other service or the UI would consume the SimpleXML directly now (or maybe ever) but it is a good debugging endpoint since it's the input to the HQMF. As the HQMF/SimpleXML generation is still somewhat of a WIP, we can tackle this during the sprint that started today

Comment (adongare) [reply src/main/java/gov/cms/madie/resources/PackageController.java]: yeah, right. this is kind of test end point for us for now. but I've addressed it now. Thanks Rohit!

File: src/main/java/gov/cms/madie/services/PackagingService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/services/PackagingService.java]: done

**Review by RohitKandimalla — APPROVED** [review]
